plotting/quicklook_generator.py

```diff
@@ -0,0 +1,391 @@
+        Parsed dataset representing the contents of the CDF file.
+    """
+    if isinstance(file_path, imap_data_access.ImapFilePath):
+        file_path = file_path.construct_path()
+
+    dataset = cdf_to_xarray(file_path, kwargs)
+
+    # cdf_to_xarray converts single-value attributes to lists
+    # convert these back to single values where applicable
+    for attribute in dataset.attrs:
+        value = dataset.attrs[attribute]
+        if isinstance(value, list) and len(value) == 1:
+            dataset.attrs[attribute] = value[0]
+
+    # Remove attributes specific to xarray plotting from vars and coords
+    # TODO: This can be removed if/when feature is added to cdf_to_xarray to
+    #      make adding these attributes optional
+    if remove_xarray_attrs:
+        for key in dataset.variables.keys():
+            for xarray_key in ISTP_TO_XARRAY_ATTRS.values():
+                dataset[key].attrs.pop(xarray_key, None)
+
+    return dataset
+
+
+def dataset_into_xarray(file_name: str) -> xr.Dataset | bool:
+    """
+    Use IMAP file name/directory structure to load a CDF file as a xr.Dataset.
+
+    Parameters
+    ----------
+    file_name : str
+        Desired file to generate quicklook from.
+
+    Returns
+    -------
+    None
+        This function returns nothing.
+    """
+    # Pulls needed info from IMAP file name structure for file_path
+    mission, instrument, level, descriptor, year_month, version_no = file_name.split(
+        "_"
+    )
+    year = year_month[:4]
+    month = year_month[4:6]
+
+    # Define file_path
+    file_path = os.path.join(
+        os.path.dirname(__file__),
+        "data",
+        "imap",
+        instrument,
+        level,
+        year,
+        month,
+    )
+    full_path = os.path.join(file_path, file_name)
+    # print('Full Path: ' + full_path)
+
+    # Check if file exists
+    if not os.path.exists(full_path):
+        print("File does not exist")
+        return False
+    else:
+        # Create xr.Dataset for plotting purposes
+        data_set = load_cdf(full_path)
+        return data_set
+
+
+def convert_j2000_to_utc(time_array: np.ndarray) -> np.ndarray:
+    """
+    Convert time from j2000 to utc.
+
+    Parameters
+    ----------
+    time_array : np.ndarray
+        Desired array to convert from j200 to UTC.
+
+    Returns
+    -------
+    np.ndarray
+        The newly converted UTC time array.
+    """
+    times = TTJ2000_EPOCH + time_array.astype("timedelta64[ns]")
+    return times
+
+
+# TODO: Check the way this is handled.
+def generate_instrument_quicklook(filename: str) -> QuicklookGenerator:
+    """
+    Determine which abstract class to use for a given file.
+
+    Parameters
+    ----------
+    filename : str
+        Desired file to generate quicklook from.
+
+    Returns
+    -------
+    QuicklookGenerator
+        Proper abstract class for file.
+    """
+    mission, instrument, level, descriptor, year_month, version_no = filename.split("_")
```

Comment:
ScienceFilePath is probably more reliable here, but also maybe overkill. 

If you aren't using the rest of the variables, you can name them `_` to indicate that:
```suggestion
    _, instrument, _, _, _, _ = filename.split("_")
```

or:
`instrument = filename.split("_")[1]`